# bindPredictML17

bindPredictML17 is a sequence-based method to predict binding site residues. 
It uses an Artifical Neural Network (ANN) and was trained on enzymes and DNA-binding proteins 
(i.e. it is most suited to predict binding sites in these type of proteins).
It uses 10 input features calculated for each residue:
- cumulative coupling scores
- cumulative coupling scores filtered by distance
- cumulative coupling scores filtered by solvent accessibility
- clustering coefficients
- clustering coefficients filtered by distance
- clustering coefficients filterd by solvent accessibility
- BLOSUM62-SNAP2 comparison
- BLOSUM62-EVmutation comparison
- relative solvent accessibility
- conservation


## Implementation
bindPredict is written in Python3. The trained model using the 10 input features named above is included in this repository.
bindPredict can be executed via command line runnning "bindPredict.py".
To calculate the different scores used for the prediction, bindPredict needs pre-calculated results from external programs.
As input parameters, the method needs:
- path to a folder containing EVcouplings results 
  (including *.di_scores, *_CouplingScoresCompared_all.csv, *.epistatic_effect, *_frequencies.csv, *_alignment_statistics.csv)
- path to a folder containing SNAP2 results
- path to a folder containing PROFphd results

The method then makes a prediction for every protein present in the SNAP2 results folder. So, all files should be named the same (e.g. P0A0I7.snap2, P0A0I7.profphd,...)
If any files are not provided, the corresponding scores are not calculated and set to 0.

## External programs
The pre-calculated results from external programs can be calculated as follows:

- **SNAP2** results: SNAP2 can be run using [predictprotein.org](https://www.predictprotein.org/) [1]. After the job has finished, the results can be downloaded under "Effect of Point Mutations" using the export button.
- **PROFphd** results: PROFphd can also be run using [predictprotein.org](https://www.predictprotein.org/). After the job has finised, the view of the results have to be changed to "text" (upper right corner) and the text part under "Secondary Structure" has to be copied to a text file.
- **EVcouplings** results (\*.di_scores, \*_CouplingScoresCompared_all.csv, \*_frequencies.csv, \*_alignment_statistics.csv): [EVcouplings](http://evfold.org/evfold-web/newmarkec.do) [2,3] is available as a Github repository. A detailed description on how to run EVcouplings can be found [here](https://github.com/debbiemarkslab/EVcouplings). EVcouplings only provides EC scores inferred by plmDCA. To calculate DI scores, one can use
  - **FreeContact** [4] which can be downloaded as a [debian package](https://packages.debian.org/search?keywords=freecontact). Using the alignment generated by EVcouplings, DI scores can be calculated using the option "evfold".
  - the **EVcouplings** [webserver](http://evfold.org/evfold-web/newmarkec.do). DI scores can be calculated by entering the UniProt identifier or the sequence in FASTA format and by choosing "DI" as coupling scoring.
- **EVmutation** results (\*.epistatic_effect): 
[EVmutation](https://marks.hms.harvard.edu/evmutation/index.html) [5] is only available as a Github repository. A detailed description on how to run EVmutation can be found [here](https://htmlpreview.github.io/?https://github.com/debbiemarkslab/EVmutation/blob/master/EVmutation.html)

If results from EVcouplings or EVmutation cannot be obtained due to problems with local installation, a minimal version of bindPredict can be run by using the pre-trained model, but only providing SNAP2 and PROFphd results as input for the prediction. All other input scores will be set to 0. However, this will limit performance of the method.

## Cite
If you are using this method and find it helpful, we would appreciate if you could cite the following publication:
Schelling, M., Hopf T.A., Rost, B. (2018). [Evolutionary couplings and sequence variation effect predict protein binding sites](https://onlinelibrary.wiley.com/doi/abs/10.1002/prot.25585). Proteins **86**: 1064-1074

## References
[1] Yachdav, G., Kloppmann, E., Laszlo, K., Hecht, M., Goldberg, T., Hamp, T., Hönigschmid, P., Schafferhans, A., Roos, M., Bernhofer, M. et al. (2014). [PredictProtein - an open resource for online prediction of protein structural and functional features](https://academic.oup.com/nar/article/42/W1/W337/2435518). Nucleic acids research **42**, pages W337–W343.

[2] Marks, D. S., Colwell, L. J., Sheridan, R., Hopf, T. A., Pagnani, A., Zecchina, R., Sander, C. (2011). [Protein 3D Structure Computed from Evolutionary Sequence Variation](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0028766). PLoS ONE **6**(12): e28766.

[3] Hopf, T. A., Schärfe, C. P. I., Rodrigues, J. P. G. L. M., Green, A. G., Kohlbacher, O., Sander, C., Bonvin, A. M. J. J., Debora S Marks, D.S. (2014) [Sequence co-evolution gives 3D contacts and structures of protein complexes](https://elifesciences.org/articles/03430). eLife; 3:e03430

[4] Kaján L., Hopf T. A., Kalaš M., Marks D. S., Rost B. (2014) [FreeContact: fast and free software for protein contact prediction from residue co-evolution.](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-15-85). BMC Bioinformatics **15**:58

[5] Hopf, T. A., Ingraham, J. B., Poelwijk, F.J., Schärfe, C.P.I., Springer, M., Sander, C., & Marks, D. S. (2017). 
[Mutation effects predicted from sequence co-variation](https://www.nature.com/articles/nbt.3769). Nature Biotechnology **35**, pages 128–135.

